### ISSUE TITLE
The game starts lagging tremendously if you move really fast.

### ISSUE THREAD

**Issue by Suvanth-Erranki** [maitian352/Meadow-Guarder#197]
![image](https://user-images.githubusercontent.com/76885647/147394804-91be5bae-5a11-4063-8f04-5c75e6fcbcf7.png)

When I move really fast and use my weapons with this setup, the game starts lagging, pretty badly.

**Comment by maitian352** [comment]
I set my maxspeed to 1000 and its not working.

**Comment by definitely-nobody-is-here** [comment]
then dont

**Comment by definitely-nobody-is-here** [comment]
If possible, add a max steps to the physics movement calculations so if you go 1000 pixels in one tick, instead of doing a thousand checks for collisions, it only does 50 and it moves in 20 pixel increments, and then when you hit a collision it backtracks to the last check and starts the fine 1-pixel increments

**Comment by maitian352** [comment]
I believe this is client side, as I set my maxspeed to 1000 and it didn't lag.

**Comment by Suvanth-Erranki** [comment]
No, it only happens if you shoot weapons while moving around, continuously changing direction.

**Comment by Suvanth-Erranki** [comment]
> I believe this is client side, as I set my maxspeed to 1000 and it didn't lag.

Yes, this is definitely client side, not server side lag.

**Comment by Suvanth-Erranki** [comment]
I didn't mean to close, it was an accident.

**Comment by maitian352** [comment]
> If possible, add a max steps to the physics movement calculations so if you go 1000 pixels in one tick, instead of doing a thousand checks for collisions, it only does 50 and it moves in 20 pixel increments, and then when you hit a collision it backtracks to the last check and starts the fine 1-pixel increments. It's probably a lot less laggy

But then no 1 pixel walls. :(

**Comment by definitely-nobody-is-here** [comment]
1 pixel walls don't make sense.
